tests/tracking/fluent/test_fluent.py

```diff
@@ -519,26 +520,81 @@ def test_get_run():
         assert run.info.user_id == "my_user_id"
 
 
+def call_search_runs():
+    return search_runs(output_format="list") if "MLFLOW_SKINNY" in os.environ else search_runs()
+
+
+def validate_search_runs(results, data):
+    if "MLFLOW_SKINNY" in os.environ:
+        result_data = defaultdict(list)
+        for run in results:
+            result_data["status"].append(run.info.status)
+            result_data["artifact_uri"].append(run.info.artifact_uri)
+            result_data["experiment_id"].append(run.info.experiment_id)
+            result_data["run_id"].append(run.info.run_id)
+            result_data["start_time"].append(run.info.start_time)
+            result_data["end_time"].append(run.info.end_time)
+
+        assert result_data == data
+    else:
```

Comment:
Re: https://github.com/mlflow/mlflow/pull/3972/files#r563478944: this if/else logic depends on the assumption that full client testing always uses `output_format = 'pandas'`, which is a bit hard to infer without comments / documentation. I think it'd be clearer if we tested both output formats in the full client & performed these kinds of branching asserts based on `output_format` rather than `os.environ["MLFLOW_SKINNY"]`.